components/mediation/mediators/dataservices-mediator/org.wso2.micro.integrator.mediator.dataservice/src/main/java/org/wso2/micro/integrator/mediator/dataservice/DataServiceCallMediator.java

```diff
@@ -0,0 +1,411 @@
+import org.apache.synapse.util.MessageHelper;
+import org.w3c.dom.Document;
+import org.wso2.micro.integrator.dataservices.core.DataServiceFault;
+import org.wso2.micro.integrator.dataservices.core.DataServiceProcessor;
+import org.xml.sax.InputSource;
+import org.xml.sax.SAXException;
+
+import javax.xml.namespace.QName;
+import javax.xml.parsers.DocumentBuilder;
+import javax.xml.parsers.DocumentBuilderFactory;
+import javax.xml.parsers.ParserConfigurationException;
+import java.io.IOException;
+import java.io.StringReader;
+import java.util.List;
+import java.util.regex.Matcher;
+
+public class DataServiceCallMediator extends AbstractMediator {
+
+    private String dsName;
+    private Operations operations;
+    public enum OperationsType {SINGLE_REQ, REQUEST_BOX, BATCH_REQ}
+    private String targetType;
+    private String propertyName;
+    private OMFactory fac;
+    private OMNamespace omNamespace;
+    private SynapseLog synLog;
+
+
+    public boolean mediate(MessageContext messageContext) {
+
+        synLog = getLog(messageContext);
+        String serviceName = getDsName();
+
+        if (synLog.isTraceOrDebugEnabled()) {
+            synLog.traceOrDebug("The invoking data service name : " + getDsName());
+        }
+        omNamespace = new OMNamespaceImpl(DataServiceCallMediatorConstants.PAYLOAD_NAME_SPACE_URI,
+                DataServiceCallMediatorConstants.PAYLOAD_PREFIX);
+        fac = OMAbstractFactory.getOMFactory();
+        try {
+            // clone the message context to append payloads to invoke dataservice
+            MessageContext cloneMessageContext = MessageHelper.cloneMessageContext(messageContext);
+            // Casting the synapse message context to axis2 message context
+            org.apache.axis2.context.MessageContext axis2MessageContext = ((Axis2MessageContext) cloneMessageContext)
+                    .getAxis2MessageContext();
+            // Get the Axis service name of the give dataservice name
+            AxisService axisService = axis2MessageContext.getConfigurationContext().getAxisConfiguration()
+                    .getService(serviceName);
+            if (axisService != null) {
+                if (synLog.isTraceOrDebugEnabled()) {
+                    synLog.traceOrDebug("The axisService :" + axisService.getName());
+                    synLog.traceOrDebug("The service name space URI : " + axisService.getTargetNamespace());
+                }
+                // Set the axis service into the axis2 message context
+                axis2MessageContext.setAxisService(axisService);
+                OMElement payload = addRootOperation(axis2MessageContext, messageContext);
+                if (axis2MessageContext.getEnvelope().getBody().getFirstElement() != null) {
+                    axis2MessageContext.getEnvelope().getBody().getFirstElement().detach();
+                }
+                axis2MessageContext.getEnvelope().getBody().addChild(payload);
```

Comment:
seems there are some issue in setting the payload back. this gives back the input request even after this.
we can try to set the envelope to axis2MsgCtx. 
```java
    public static void setPayloadInEnvelope(MessageContext axis2MsgCtx, OMElement payload)
            throws ContentBuilderException {

        JsonUtil.removeJsonPayload(axis2MsgCtx);
        try {
            axis2MsgCtx.setEnvelope(TransportUtils.createSOAPEnvelope(payload));
        } catch (AxisFault e) {
            throw new ContentBuilderException("Failed to set XML content.", e);
        }
    }
```